pkg/rpc/response/types.go

```diff
@@ -24,13 +26,38 @@ type Raw struct {
 	Result json.RawMessage `json:"result,omitempty"`
 }
 
+// AbstractResult is an interface which represents either single JSON-RPC 2.0 response
+// or batch JSON-RPC 2.0 response.
+type AbstractResult interface {
+	RunForErrors(req *request.Request, f func(r *request.In, jsonErr *Error))
```

Comment:
`request.In` is not really needed here, it's in the context of `RunForErrors` invoker, so a closure solves this problem.